### PR TITLE
rlite: init at 2020-03-31

### PR DESCRIPTION
**Pull request by Twey**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dasj19** [comment]
Interesting project. Just tested with `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 92241"`

Results:
```

error: build of '/nix/store/jgahhaw0hmrymsvdyik1fm401zk81mgy-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/92241
2 packages marked as broken and skipped:
linuxPackages_hardkernel_4_14.rlite linuxPackages_hardkernel_latest.rlite

6 packages failed to build:
linuxPackages_5_6.rlite linuxPackages_5_7.rlite linuxPackages_latest-libre.rlite linuxPackages_latest_hardened.rlite linuxPackages_latest_xen_dom0.rlite linuxPackages_latest_xen_dom0_hardened.rlite

10 packages built:
linuxPackages-libre.rlite linuxPackages.rlite linuxPackages_4_14.rlite linuxPackages_4_19.rlite linuxPackages_4_4.rlite linuxPackages_4_9.rlite linuxPackages_hardened.rlite linuxPackages_testing_bcachefs.rlite linuxPackages_xen_dom0.rlite linuxPackages_xen_dom0_hardened.rlite
```

File: pkgs/os-specific/linux/rlite/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+, protobuf
+, cmake
+, python
+, swig
+, wpa_supplicant
+, hostapd
+, enable-python ? false
+, enable-wifi ? false
+}:
+let
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:15]: Please remove let in and move it into rec.

File: pkgs/os-specific/linux/rlite/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  deps = [ which cmake protobuf ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:16]: Remove this and move it to the correct location.

File: pkgs/os-specific/linux/rlite/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  meta = with stdenv.lib; {
+    description = "A light RINA implementation";
+    homepage    = https://github.com/rlite/rlite;
+    license     = licenses.lgpl21;
+    maintainers = [ maintainers.twey ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:21]:
```suggestion
    maintainers = with maintainers; [ twey ];
```

File: pkgs/os-specific/linux/rlite/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  pname = "rlite";
+  version = "2020-03-31";
+  src = fetchFromGitHub {
+    owner = "rlite";
+    repo = pname;
+    rev = "0612ddd2423bb00225a573e78194ed7eaafc5604";
+    sha256 = "1gngcxxl792lqpwdvxqzlajgx3rgxaksxdbj55hjcr4d3qiqq905";
+  };
+
+  buildInputs = stdenv.lib.concatLists [
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:34]: Please use ++ instead of concatLists.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by dasj19** [comment]
This is waiting for some time... can you apply Sandro's comments so we can move on to approval and merge? Thanks.

File: pkgs/os-specific/linux/rlite/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  ];
+
+  postUnpack = "patchShebangs .";
+  configurePhase = ''
+    ./configure \
+      --prefix $out \
+      --kernbuilddir "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" \
+  '';
+  buildPhase = "make -j$NIX_BUILD_CORES";
+  installPhase = "make install";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:42]: Those should be done automatically. Please give removing this a try.

File: pkgs/os-specific/linux/rlite/default.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  buildInputs = [ which cmake protobuf ]
+    ++ stdenv.lib.optionals enable-python [ python swig ]
+    ++ stdenv.lib.optionals enable-wifi [ wpa_supplicant hostapd ];
+
+  patches = [
+    ./configure-remove-hardcoded-paths.patch
+    ./strip-hardcoded-usr-install-prefix.patch
+  ];
+
+  postUnpack = "patchShebangs .";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:35]: This should be postPatch. Which files exactly?

File: pkgs/os-specific/linux/rlite/default.nix [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  src = fetchFromGitHub {
+    owner = "rlite";
+    repo = pname;
+    rev = "0612ddd2423bb00225a573e78194ed7eaafc5604";
+    sha256 = "1gngcxxl792lqpwdvxqzlajgx3rgxaksxdbj55hjcr4d3qiqq905";
+  };
+
+  buildInputs = [ which cmake protobuf ]
+    ++ stdenv.lib.optionals enable-python [ python swig ]
+    ++ stdenv.lib.optionals enable-wifi [ wpa_supplicant hostapd ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/rlite/default.nix:28]:
```suggestion
    ++ lib.optionals enable-python [ python swig ]
    ++ lib.optionals enable-wifi [ wpa_supplicant hostapd ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Twey** [comment]
Sorry, I ran out of time to work on this, and since the upstream project has stagnated it no longer compiles on recent Linux.  I'm closing this PR.

**Pull request closed without merging**